AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine+DeleteTransaction.swift

```diff
@@ -11,9 +11,43 @@ import Foundation
 import AWSPluginsCore
 
 extension StorageEngine {
+
+    enum DeleteInput {
+        case withId(id: Model.Identifier, predicate: QueryPredicate?)
+        case withPredicate(predicate: QueryPredicate)
+
+        /// Returns a computed predicate based on the type of delete scenario it is.
+        var predicate: QueryPredicate {
+            switch self {
+            case .withId(let id, let predicate):
+                if let predicate = predicate {
+                    return field("id").eq(id).and(predicate)
+                } else {
+                    return field("id").eq(id)
+                }
+            case .withPredicate(let predicate):
+                return predicate
+            }
+        }
+
+        /// Returns the `id` if the delete input is by `id` and contains `predicate`, return `nil` otherwise
+        var isDeleteByIdWithPredicate: Model.Identifier? {
```

Comment:
nit: the property name sounds like an assertion but doesn't return a boolean maybe we can just call it `id` or `modelIdentifier`?

Reply:
no longer needed due to having the 3 cases